lib/io/can.cpp

```diff
@@ -94,4 +96,24 @@ std::optional<CanFrame> Can::receive()
   return received_message;
 }
 
+void Can::listen()
+{
+  std::optional<CanFrame> data;
+  if (ioctl(socket_, FIONREAD) >= sizeof(CanFrame)) { data = receive(); }
+  if (data == std::nullopt) { return; }
```

Comment:
I don't like the logic here.

```cpp
if (ioctl(socket_, FIONREAD) < sizeof(CanFrame)) { return; }
const auto data = receive();
if (!data) {
  return;
}
```

Reply:
Also, this looks like it can succeed and fail? Maybe you should have a non-void return type here to indicate what happened to the caller.